### ISSUE TITLE
kbox pull stopped working with Pantheon app instance. kbox create pantheon stopped working Error: Processing failed.

### ISSUE THREAD

**Issue by pi-ron** [kalabox/kalabox#369]
All of a sudden (it seems) kbox pull stopped working for a particular app.  The only thing I can think of that was different to the previous successful pull was that I was connected to the database with Sequel Pro, not sure if that would of caused this failure.

``````
$ kbox pull
? Also refresh from latest database backup? Yes
? Also grab latest files? No

#Lots of normal docker output, then:

Error: Processing failed.
Error: Error: Processing failed.
    at /Users/RoamingMac/Sites/tfse/plugins/kalabox-plugin-pantheon/lib/client.js:209:13
From previous event:
    at /Users/RoamingMac/Sites/tfse/plugins/kalabox-plugin-pantheon/lib/client.js:207:11
From previous event:
    at Client.__request (/Users/RoamingMac/Sites/tfse/plugins/kalabox-plugin-pantheon/lib/client.js:192:4)
    at Client.getBackups (/Users/RoamingMac/Sites/tfse/plugins/kalabox-plugin-pantheon/lib/client.js:375:15)
    at Terminus.hasDBBackup (/Users/RoamingMac/Sites/tfse/plugins/kalabox-plugin-pantheon/lib/terminus.js:291:19)
    at engine.isRunning.then.then.then.then.then.then.payload (/Users/RoamingMac/Sites/tfse/plugins/kalabox-plugin-pantheon/lib/pull.js:129:27)
    at process._tickCallback (node.js:419:13)
``````

**Comment by pi-ron** [comment]
I have tried shutting down kbox and restarting.  I've also tried this across multiple apps, getting the same error for all pantheon apps.

**Comment by pi-ron** [comment]
Tried restarting comp too https://www.youtube.com/watch?v=p85xwZ_OLX0

**Comment by pi-ron** [comment]
OK, I tried pulling a new existing site down from pantheon and got the same error.  Looks like an issue with git maybe, this is the final output before the error:

``````
info: DOCKER =>  Ad hoc container started. 880df2e15811bcfb08ce4ebe451e57c4a8906c02cfda55de0740e5e94333ebd3
Identity added: /ssh/.ssh/id_rsa (/ssh/.ssh/id_rsa)
fatal: Not a git repository (or any parent up to mount point /code)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
Error while installing app: "di-business-package": Error: Processing failed.
Error: Error: Processing failed.
    at /Users/RoamingMac/Sites/di-business-package/plugins/kalabox-plugin-pantheon/lib/client.js:209:13
From previous event:
    at /Users/RoamingMac/Sites/di-business-package/plugins/kalabox-plugin-pantheon/lib/client.js:207:11
From previous event:
    at Client.__request (/Users/RoamingMac/Sites/di-business-package/plugins/kalabox-plugin-pantheon/lib/client.js:192:4)
    at Client.getBackups (/Users/RoamingMac/Sites/di-business-package/plugins/kalabox-plugin-pantheon/lib/client.js:375:15)
    at Terminus.hasDBBackup (/Users/RoamingMac/Sites/di-business-package/plugins/kalabox-plugin-pantheon/lib/terminus.js:291:19)
    at engine.isRunning.then.then.then.then.then.then.payload (/Users/RoamingMac/Sites/di-business-package/plugins/kalabox-plugin-pantheon/lib/pull.js:129:27)
    at process._tickCallback (node.js:419:13)
``````

**Comment by labboy0276** [comment]
I am seeing the original error on this ticket all of a sudden today:

```
info: DOCKER =>  Ad hoc container started. d73f75d5e176906b5b64074c1afc3ccadb211fd5fd7922540792ff5638df7091
Error while installing app: "cor-777": Error: Processing failed.
Error: Error: Processing failed.
    at /home/john/kalabox/apps/cor-777/plugins/kalabox-plugin-pantheon/lib/client.js:209:13
From previous event:
    at /home/john/kalabox/apps/cor-777/plugins/kalabox-plugin-pantheon/lib/client.js:207:11
From previous event:
    at Client.__request (/home/john/kalabox/apps/cor-777/plugins/kalabox-plugin-pantheon/lib/client.js:192:4)
    at Client.getBackups (/home/john/kalabox/apps/cor-777/plugins/kalabox-plugin-pantheon/lib/client.js:375:15)
    at Terminus.hasDBBackup (/home/john/kalabox/apps/cor-777/plugins/kalabox-plugin-pantheon/lib/terminus.js:291:19)
    at /home/john/kalabox/apps/cor-777/plugins/kalabox-plugin-pantheon/lib/pull.js:129:27
    at processImmediate [as _immediateCallback] (timers.js:367:17)
```

I tried removing ~/.terminus and that did not help.  I am gonna douche the install and try again.

**Comment by labboy0276** [comment]
PS this is happening on every pantheon app I am trying to install.

**Comment by labboy0276** [comment]
I completely douched my system (including removing all ssh keys for b2d, ~/.kalabox && ~/.terminus folders)

I am still getting error above.  Something is a miss with terminus I am guessing since this worked fine 3 hours ago....

**Comment by labboy0276** [comment]
Based on my debugging, the REST _request is just flat out failing now.  I forget how to check to see if Terminus is down or up.  I know Pantheon itself had a slew of problems yesterday due to the major outage on the east coast.  However, that doesn't explain that it worked fine this AM and didn't after that....

GAH

**Comment by reynoldsalec** [comment]
I'm unable to pull down apps as well, noticing the following error when I attempt to do so:

```
ssh: Could not resolve hostname [codeserver.dev.Warning: There is a newer version of Terminus (0.6.1) avaiable
```

It appears that some Terminus warning text is getting concatenated inappropriately into the hostname of the codeserver, which is causing the inability to pull down the code.

@labboy0276 is trying some "ninja magic".

**Comment by reynoldsalec** [comment]
Code where we're creating codeserver URL (where this error is presumably coming from): https://github.com/kalabox/kalabox-app-pantheon/blob/master/app/plugins/kalabox-plugin-pantheon/lib/pull.js#L78

In the future we might have an issue with the Terminus API endpoints changing from "dashboard.getpantheon.com" to "dashboard.pantheon.io", might need to fix here: https://github.com/kalabox/kalabox-app-pantheon/blob/master/app/plugins/kalabox-plugin-pantheon/lib/client.js

**Comment by reynoldsalec** [comment]
Ok, the basic parsing of the UUID for the codeserver URL should be fixed by changes @bcauldwell made on the issue369.1 branch on kalabox-app-pantheon, but that has produced further errors, most of the messages revolving around this one:

```
Error while installing app: "somemsdw": EVENTS =>  Error dispatching event listeners [post-install].
VError: Error running ad hoc container with kalabox/terminus:v0.9.0.: PHP Fatal error:  Call to a member function error() on a non-object in phar:///usr/local/bin/terminus/php/class-terminus.php on line 221
```

There is a relevant issue for this error on pantheon-systems/cli#198, but I think this is a red herring, seeing as we haven't had any issues like this prior to today and, to my knowledge, have php5-curl installed.

I'm going to follow up by first trying @labboy0276's Terminus upgrade in conjunction with @bcauldwell's fixes. If that isn't a slamdunk, I'll go back to debugging with the older version of Terminus (which we know worked).

**Comment by reynoldsalec** [comment]
@joshkoenig, perhaps you or someone over at Pantheon know if anything changed with Terminus's API responses within the last 24hrs? We started getting back a "Warning:" message regarding a new version of Terminus being available alongside the site UUID when running "terminus site info --json --site=somesite --field=id" on Terminus 5.5, maybe this has been fixed in 6.0 but I would think the --json response shouldn't include those types of outputs, since I'm guessing most people consuming responses as JSON will be using Terminus programmatically as an API more than a CLI tool...

**Comment by reynoldsalec** [comment]
Ok folks, with @bcauldwell's fix and by deleting the "Warning:" message from Terminus 5.5 (see https://github.com/pantheon-systems/cli/blob/0.5.x/php/terminus.php#L34) I got this to work, will figure out how to role a fix for everyone tomorrow, thanks for your patience!

**Comment by pi-ron** [comment]
Thanks mate great work!

**Comment by labboy0276** [comment]
Just to repeat what I put in our slack so the world can see.  I went a different route and did the following:

after speaking with @pirog, I went a different route:

1. I forked and tried to create my own phar that removes that stupid warning code.  I dont think the phar built right as I have never done that before and so I went with the next option:
2. I rolled this back to 0.5.3 before the commit of said warning message

2 is working now and if anyone needs it, it is issue369.2 branch on the pantheon app. I will await @reynoldsalec to see if we can either fix my phar and or roll with the 0.5.3 until @pirog has time to fix this from mother russia.

**Comment by reynoldsalec** [comment]
Pushed up my changes in kalabox/kalabox-app-pantheon#10, @labboy0276 if you want to test that you'll need to pull it down and build locally.

**Comment by joshkoenig** [comment]
Might be related to this?

https://github.com/pantheon-systems/cli/issues/315

On Wed, Aug 12, 2015 at 8:55 AM, reynoldsalec <notifications@github.com>
wrote:

> Pushed up my changes in kalabox/kalabox-app-pantheon#10
> <https://github.com/kalabox/kalabox-app-pantheon/pull/10>, @labboy0276
> <https://github.com/labboy0276> if you want to test that you'll need to
> pull it down and build locally.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/369#issuecomment-130353538>.
>



-- 

Josh Koenig
Co-Founder / Head of Product
josh@pantheon.io

**Comment by reynoldsalec** [comment]
Ok, merged in kalabox/kalabox-app-pantheon#10 and the kalabox-app-pantheon container has rebuilt on Docker hub, so doing a [normal update](https://github.com/kalabox/kalabox/wiki/Updating-Kalabox) should get everything working again @pi-ron. Please re-open this if that doesn't seem to work.
